pkg/repository/v1/olap.go

```diff
@@ -1333,3 +1334,32 @@ func (r *OLAPRepositoryImpl) GetTaskTimings(ctx context.Context, tenantId string
 
 	return tasksWithData, idsToDepth, nil
 }
+
+func (r *OLAPRepositoryImpl) BulkCreateEventsAndTriggers(ctx context.Context, events []sqlcv1.BulkCreateEventsParams, triggers []sqlcv1.BulkCreateEventTriggersParams) error {
+	tx, err := r.pool.Begin(ctx)
```

Comment:
If you trace up the context, this looks like it's just using `context.Background` and there's no timeout on this insert, which can be an issue (we need timeouts on all database calls). I think it's also useful to call `sqlchelpers.PrepareTx` with a statement timeout for any writes.